### ISSUE TITLE
SetHome positional settings.

### ISSUE THREAD

**Issue by PlayingWithPi** [espidev/ProtectionStones#270]
`/ps sethome` / `/ps home [name]` requires directional settings.
Currently doesn't matter what happens, the user always ends up facing South.

Furthermore, is it possible to add a [setname (optional)] to `/ps sethome` and have it update the `/ps home` list as well?
(currently, the `/ps home [name]` shows both the original home name, and the `/ps name` name as well.
TL;DR duplications occuring in `/ps home (tab)` showing both the default name & setName.

**Comment by espidev** [comment]
> Furthermore, is it possible to add a [setname (optional)] to /ps sethome and have it update the /ps home list as well?
(currently, the /ps home [name] shows both the original home name, and the /ps name name as well.
TL;DR duplications occuring in /ps home (tab) showing both the default name & setName.

I'm not sure if I understand this. What is setname? Does it have the same purpose as `/ps setname [name]`?

**Comment by Amelia-exe** [comment]
I think @PlayingWithPi means:
**/ps sethome [setName(Optional)]** is the command they'd like to see.

Correct me if I got this wrong (:

And I've noticed it too, both the default name and the new name set via:
**/ps name [name]** command is shown in:
**/ps home** (tab auto-complete)

**Comment by espidev** [comment]
I think I might understand now, you want the ability to set multiple homes for a region?

**Comment by Amelia-exe** [comment]
Request One:
To be able to set the home name, whilst doing the /ps sethome command. 

+--------------------+

Noticed an issue:

/ps home [name] -
Currently shows : 
ps-1x2y3z (default /ps home location)
Home1 (/ps name Home1 location)

They both lead back to the same place, but both are shown within /ps home command. 

The logic should be :

If ps.name == true {
    return ps.home.getUserSetName
} else return ps.home.defaultName

Not :
ps.name == true {
    return ps.home.getUserSetName && ps.home.defaultName
}

The code above is pseudo code.

**Comment by espidev** [comment]
> Request One:
To be able to set the home name, whilst doing the /ps sethome command.

That requires a rework of how PS stores homes entirely, so maybe not anytime soon

> Noticed an issue:

Are you referring to how it says `name (id)`? That is intentional to prevent confusion.

**Comment by espidev** [comment]
Directional settings were added in the latest update.

**Comment by Amelia-exe** [comment]
Yeah I saw it in the release notes ! That's awesome (:

**Comment by Amelia-exe** [comment]
> > Noticed an issue:
> 
> 
> 
> Are you referring to how it says `name (id)`? That is intentional to prevent confusion.

No. Within the tab-autocomplete of /ps home [homeName], it'll state both Home1 (the name that was set) and the ps-1x2y3z (the default name), as if they're TWO SEPARATE LOCATIONS.

**Comment by espidev** [comment]
> No. Within the tab-autocomplete of /ps home [homeName], it'll state both Home1 (the name that was set) and the ps-1x2y3z (the default name), as if they're TWO SEPARATE LOCATIONS.

Oh, you're talking about autocomplete. That's intentional, if a user wanted to use the id instead of the name (since they both work).

**Comment by Amelia-exe** [comment]
> > No. Within the tab-autocomplete of /ps home [homeName], it'll state both Home1 (the name that was set) and the ps-1x2y3z (the default name), as if they're TWO SEPARATE LOCATIONS.
> 
> 
> 
> Oh, you're talking about autocomplete. That's intentional, if a user wanted to use the id instead of the name (since they both work).

I mean.... if the user has gone out of their way to set a name... why would they want to see the old one?

Looks like clutter at the moment imho

**Comment by espidev** [comment]
https://ci.espi.dev/job/ProtectionStones/363/

**Comment by Amelia-exe** [comment]
You, Mister, are an amazing human being :) !!!

**Comment by Amelia-exe** [comment]
Btw.... 

You could put it to the community, how they want to see it? 

Like a Poll page or something like that?

Would be nice to have user feedback to see how they'd like to see commands looking like in the end ? @espidev

**Comment by espidev** [comment]
There's no way to know until I get complaints, polls generally aren't very effective

**Comment by Amelia-exe** [comment]
That sucks, anyways, I guess that this issue is now resolved ?

**Comment by PlayingWithPi** [comment]
Issue has been resolved.
Thanks for the update @espidev 

sorry I’ve not been very active but it seems Amelia has the same thoughts !
